obsidian-multiplatform-highlights-import plugin

wwwkieran opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/wwwkieran/obsidian-multiplatform-highlights-import

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [ x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files
  - [x ] `main.js`
  - [x ] `manifest.json`
  - [ x] `styles.css` _(optional)_
- [x ] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x ] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x ] My README.md describes the plugin's purpose and provides clear usage instructions.
- [ x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [ x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x ] I have added a license in the LICENSE file.
- [x ] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[Copyright (c) 2022 Bitwise](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/LICENSE#L3)
Update this, and please give credit to the original sources of the code used.

["minAppVersion": "0.15.9",](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/manifest.json#L5)
You are using newer API's, set this to the latest public build number.

[const KoboHighlightsImporterIconEl = this.addRibbonIcon('e-reader', 'Import Book Highlights', () => {](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/main.ts#L17), [.setName('Template Path')](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L126)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[import {sanitize} from "sanitize-filename-ts";](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/aggregator.ts#L5), [const sanitizedBookName = sanitize(book.book.title)](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/aggregator.ts#L33)
Normalize path already does the sanitization, no need to do it twice.

[import moment from "moment/moment";](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/aggregator.ts#L3)
Use `import {moment} from "obsidian";` instead.

[Suggest.ts](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/suggestors/Suggest.ts#L1-L213)
This should no longer be used, implement a `AbstractInputSuggest` of copy/pasting this code.

[this.containerEl.createEl('h2', { text: this.plugin.manifest.name });](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L42)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[this.containerEl.createEl('h3', { text: "Extractor Settings" })](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L44), [this.containerEl.createEl('h3', { text: "Output Settings" })](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L49)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)


[this.containerEl.createEl('h3', { text: "Extractor Settings" })](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L44), [this.containerEl.createEl('h3', { text: "Output Settings" })](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L49), [this.containerEl.createEl('h4', { text: "Kobo" })](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L62-L63), [this.containerEl.createEl('h4', { text: "Apple Books" })](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/settings/Settings.ts#L95)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[import { base64 } from "./sql-wasm-base64";](https://github.com/wwwkieran/obsidian-multiplatform-highlights-import/blob/8945a791710b65e37afe25e445ed52eac504871e/src/binaries/sql-wasm.ts#L1-L3)
Please use the proper mechanism to load the WASM binaries, and don't use base64 encoding here.

wwwkieran commented:
@joethei I have addressed all your review comments in release 0.1.1, except for `Please use the proper mechanism to load the WASM binaries, and don't use base64 encoding here.` I have not been successful in loading raw .wasm files in my plugin. If there is an idiomatic way to do this, could you please link it? Otherwise, can you please approve this approach which is this same approach used in [obsidian-kobo-highlights-import](https://github.com/OGKevin/obsidian-kobo-highlights-import) which was previously approved by you.

joethei commented:
I never reviewed that plugin, only the change in the maintainer.
Also the base64 encoded part was in the plugin when it was reviewed initially.

To load WASM binaries there is a esbuild plugin that should be used.

wwwkieran commented:
Ok, I was able to get it working using esbuild as you suggested. FWIW, when esbuild loads in a binary it just Base64 [encodes it inline and then decodes it](https://esbuild.github.io/content-types/#binary) when the program runs, so it is effectively the same as it was before...

I also had to re-add the filename sanitization. `normalizePath` does not remove colons from filenames. 

Anyway, all the changes are now ready to go with version 0.1.2.

joethei commented:
The reason why I asked you to use the esbuild plugin is because it makes it easier to read and validate the source of the WASM code.
